assets/src/scripts/lib/redux.js

```diff
@@ -0,0 +1,86 @@
+// This code is derived from d3-redux: https://github.com/couchand/d3-redux
```

Comment:
Why not just use npm install this?

Reply:
I made modifications to it. Andrew and I discussed last week whether or not it's worth forking and maintaining a separate copy, but decided that given how simple it is, it makes more sense to just keep our own copy. I did offer suggestions for improvement to the original author, but he hasn't responded yet.

Reply:
What kind of changes?

Reply:
The author intended it to only be used via its fromState function with D3's data binding (data and datum functions). So to make it more general, I updated the connect function to pass the state, and added the link function to optimize the redraws.